src/app-layer-ssl.c

```diff
@@ -143,11 +146,101 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, uint8_t *input,
     switch (ssl_state->curr_connp->handshake_type) {
         case SSLV3_HS_CLIENT_HELLO:
             ssl_state->flags |= SSL_AL_FLAG_STATE_CLIENT_HELLO;
+
+            /* skip version */
+            input += SSLV3_CLIENT_HELLO_VERSION_LEN;
+
+            /* skip random */
+            input += SSLV3_CLIENT_HELLO_RANDOM_LEN;
+
+            if ((input - initial_input) > input_len)
+                goto end;
+
+            /* skip session id */
+            uint8_t session_id_length = *(input++);
+
+            input += session_id_length;
+
+            if ((input + 1 - initial_input) > input_len)
```

Comment:
Still not quite happy with the pattern here. Really the > (opposed to >= ) would guarantee at least 1 byte, then the +1 another. As we know we will need 2 bytes, this works. However I think the code readability would much improve from making the '2' explicit.

Thinking more, I think it might be good to use a (local) macro. And then instead of this line, use something like:
<pre>
if (!(HAS_SPACE(2)))
    goto end;
</pre>